### ISSUE TITLE
Detail Card

### ISSUE THREAD

**Issue by nyssance** [decaf-dev/obsidian-vault-explorer#40]
I have migrated from other note apps to obsidian. I search a long time for a plugin that allows me to preview my notes just like in other apps. Now I find your project, which is very close to what I want. If you could provide a card with no spacing or very small spacing, displaying a bit of content, similar to Bear/Apple Notes/Evernote, and user pin it to the left sidebar, it would perfectly replicate the structure of other note-taking apps.
<img width="1417" alt="截屏2024-05-29 02 14 57" src="https://github.com/decaf-dev/obsidian-vault-explorer/assets/1937955/3e1ccf1a-dbd7-4cd7-b6c2-4bf7a17e71ec">

**Comment by decaf-dev** [comment]
I really like those cards! I will look into this. Thank you for the suggestion.

**Comment by javiavid** [comment]
Maybe this will help you create it.
https://github.com/xhuajin/obsidian-file-preview

I would love to see this option of showing some lines of content within your plugin, for me it is what I need to use it.

**Comment by decaf-dev** [comment]
> Maybe this will help you create it. https://github.com/xhuajin/obsidian-file-preview
> 
> I would love to see this option of showing some lines of content within your plugin, for me it is what I need to use it.

That is helpful! Thanks

**Comment by Moyf** [comment]
https://github.com/daledesilva/obsidian_project-browser

Just remembered there is another project that also has similar functionality, maybe it will be helpful.

**Comment by decaf-dev** [comment]
Added in 1.16.0

**Comment by Moyf** [comment]
> Added in 1.15.0

AMAZING——

**Comment by Moyf** [comment]
> > Added in 1.15.0
> 
> AMAZING——

How to turn on it?
I cannot see the detail (content preview) in my Grid view, version: 1.16.0

**Comment by Moyf** [comment]
> How to turn on it? I cannot see the detail (content preview) in my Grid view, version: 1.16.0

Just activated my Premium version and found out it's the new **Feed** view  :D

And I think maybe we need some kind of **Content Filter** and **Streaming Loading**?

![image](https://github.com/decaf-dev/obsidian-vault-explorer/assets/4013062/2b732968-3dd5-435b-9765-df234f785cdf)
My personal Note's concept is a H1 same with the Filename in the front.
So I can only view the Header lol

I think maybe only view the *Normal Text Content* (Excludes table / callout / headers / ... ) as an optional ?

---

And it seems now when switching into the Feed View, it would load all the notes at a time, which will cause a Freeze.

I don't know is it possible but could it be loaded "streamly"? Like when I view the first 25 notes, just loading 'em.
And when I scroll down, load the next notes.

It might be loaded several times, but perhaps separating loading can make the perception of lag less obvious? 🤔

**Comment by decaf-dev** [comment]
Are you using 1.17.1?

**Comment by decaf-dev** [comment]
Great! Glad to hear.

**Comment by nyssance** [comment]
I have same issue with @Moyf
My personal Note's also concept is a H1 same with the Filename in the front.
And some content such as banner before H1 but not in frontmatter.
So I want just show text after first H1

**Comment by decaf-dev** [comment]
@Moyf @nyssance Thanks for the feedback. I will add these as tasks.

**Comment by Moyf** [comment]
> @Moyf @nyssance Thanks for the feedback. I will add these as tasks.

Oh since @nyssance mentioned the right-click menu, the other day when I was communicating with another plugin developer, he mentioned [ModEvents](https://github.com/karstenpedersen/obsidian-daily-note-navbar/issues/11#issuecomment-2194100366) which can add several common opening methods to the right-click menu. Perhaps VE can also consider incorporating it?
![image](https://github.com/decaf-dev/obsidian-vault-explorer/assets/4013062/0d889b5e-f85c-4b80-a171-c324691f1196)

**Comment by nyssance** [comment]
If modevents is an good plugin, maybe you can recommend each other.、

And a new feature after I installed 1.18.
It's display 250 characters in feed view now. My notes by English/Chinese, a Chinese character is approximately twice the width of an English character, the height of the feed card is too high, and only three cards can be displayed on the screen. Perhaps a configuration option to set the feed height would be better, but I prefer not to configure it by character length. This solution would require converting character width from a development perspective, which is too complex. 

Additionally, I would greatly appreciate it if the feed card could display the first image from the note (e.g., in a rectangle on the right side).

**Comment by decaf-dev** [comment]
@nyssance Please update to 1.21.2. I am looking for feedback on rendering the new line characters.

**Comment by nyssance** [comment]
> @nyssance Please update to 1.21.2. I am looking for feedback on rendering the new line characters.
> 
> EDIT: it sounds like you want things as compact as possible?

Yes, after update to newest, now just show 1.2 feed per page。The feed height is too high.

**Comment by decaf-dev** [comment]
Okay. I will add settings to change this

**Comment by decaf-dev** [comment]
@nyssance Okay. I have added a setting called `Collapse content` that defaults to true. This will replace all new line characters with a single line break. I have also replaced the 250 character trimming with the `-webkit-line-clamp` property in css. This should make the correct measurement for unicode characters like Chinese.

Here is a description of the current line clamp behavior:
| Screen size | Class name applied | Number of lines |
| < 600px | `.vaultexplorer-feed-card__content--sm` | 2 |
| >= 600px && < 1024px | `.vaultexplorer-feed-card__content--md | 3 |
| >= 1024px | `.vaultexplorer-feed-card__content--lg` | 5 |

Please upgrade to 1.24.0 and let me know how it goes.

I will add an option for filtering markdown headers. And after that an attachment display similar to Bear notes. Thank you for the screenshots. They have been very helpful.

**Comment by nyssance** [comment]
Perfect~

**Comment by decaf-dev** [comment]
@nyssance Added clickable content in 1.25.0. Will be enabled by default

**Comment by decaf-dev** [comment]
@nyssance @Moyf Okay. 1.27.0 includes a ton of updates for the feed view. A majority of Obsidian markdown is not being removed from the feed, including the first level headers (H1), so it looks a lot cleaner. I have also added settings for line clamping. The `Collapse content` setting has been changed to `Collapse style`. By default, everything is completely collapsed now with no new line characters, but you can set it to how it was in 1.25.0, with 1 new line character between every thing.

Please let me know what you think! :)

**Comment by nyssance** [comment]
@decaf-dev Big Thanks for your amazing work. I think it is nearly perfect. If there is any improvement, it would be to hide all links, such as http://xxx, [xxx](https://xxx.com), ![image](xxx.jpg),these links are very long, but users click the feed to enter the note itself and do not need to jump to this link. They do not contain useful information.

**Comment by nyssance** [comment]
Additionally, there's a small issue regarding tags. If a tag contains subtags, such as #aaa/bbb, a better approach would be to display only #bbb. However, I think this feature should be implemented by the Obsidian team.

**Comment by Moyf** [comment]
> @nyssance @Moyf Okay. 1.27.0 includes a ton of updates for the feed view. A majority of Obsidian markdown is not being removed from the feed, including the first level headers (H1), so it looks a lot cleaner. I have also added settings for line clamping. The `Collapse content` setting has been changed to `Collapse style`. By default, everything is completely collapsed now with no new line characters, but you can set it to how it was in 1.25.0, with 1 new line character between every thing.
> 
> Please let me know what you think! :)

AMAZING!! Really love it  <3
Now I can easily review many notes in Vault Explorer, really good experience !
Good Job!

**Comment by Moyf** [comment]
After experiencing it, there are two things: 
1. When switching back from other tabs, there will be obvious lag;
2. I will use more than 3 tick characters (\`), but they will not be cleaned up properly. It is recommended to use "≥ 3 tick code blocks" for deletion (see picture).
![image](https://github.com/user-attachments/assets/5186a84c-7761-4321-a6e0-791ab9bc92b2)

---

Also, I hope that the Grid view supports displaying the first image in the note, not just the image link in the `image` attribute. 
And I hope that images can be set to display in a specific way: constrain(maintain aspect ratio) or fill (horizontally tile):
![image](https://github.com/user-attachments/assets/c9746239-6186-4a4c-aa0a-816149a57abc)

![image](https://github.com/user-attachments/assets/6099605c-3b2a-41ff-b63f-5bacf111489e)

---

Additionally, I would like to provide an alternative option to display using `title` instead of the file name. But this one is not very important I think... 🤔
